stdlib/socket/src/main/java/org/ballerinalang/stdlib/socket/compiler/SocketCompilerPlugin.java

```diff
@@ -0,0 +1,173 @@
+ * under the License.
+ */
+
+package org.ballerinalang.stdlib.socket.compiler;
+
+import org.ballerinalang.compiler.plugins.AbstractCompilerPlugin;
+import org.ballerinalang.compiler.plugins.SupportedResourceParamTypes;
+import org.ballerinalang.model.tree.AnnotationAttachmentNode;
+import org.ballerinalang.model.tree.ServiceNode;
+import org.ballerinalang.util.diagnostic.DiagnosticLog;
+import org.wso2.ballerinalang.compiler.semantics.model.types.BArrayType;
+import org.wso2.ballerinalang.compiler.semantics.model.types.BRecordType;
+import org.wso2.ballerinalang.compiler.semantics.model.types.BStructureType;
+import org.wso2.ballerinalang.compiler.semantics.model.types.BType;
+import org.wso2.ballerinalang.compiler.tree.BLangFunction;
+import org.wso2.ballerinalang.compiler.tree.BLangSimpleVariable;
+
+import java.util.List;
+
+import static org.ballerinalang.model.types.TypeKind.ARRAY;
+import static org.ballerinalang.model.types.TypeKind.OBJECT;
+import static org.ballerinalang.model.types.TypeKind.RECORD;
+import static org.ballerinalang.stdlib.socket.SocketConstants.RESOURCE_ON_ACCEPT;
+import static org.ballerinalang.stdlib.socket.SocketConstants.RESOURCE_ON_CLOSE;
+import static org.ballerinalang.stdlib.socket.SocketConstants.RESOURCE_ON_CONNECT;
+import static org.ballerinalang.stdlib.socket.SocketConstants.RESOURCE_ON_ERROR;
+import static org.ballerinalang.stdlib.socket.SocketConstants.RESOURCE_ON_READ_READY;
+import static org.ballerinalang.util.diagnostic.Diagnostic.Kind.ERROR;
+
+/**
+ * Compiler plugin for validating Socket services.
+ *
+ * @since 0.985.0
+ */
+@SupportedResourceParamTypes(
+        expectedListenerType = @SupportedResourceParamTypes.Type(packageName = "socket", name = "Listener"),
+        paramTypes = { @SupportedResourceParamTypes.Type(packageName = "socket", name = "Client") }
+)
+public class SocketCompilerPlugin extends AbstractCompilerPlugin {
+
+    private static final String INVALID_RESOURCE_SIGNATURE = "Invalid resource signature for %s in service %s. ";
+    private DiagnosticLog diagnosticLog = null;
+    private int resourceCount = 0;
+
+    @Override
+    public void init(DiagnosticLog diagnosticLog) {
+        this.diagnosticLog = diagnosticLog;
+    }
+
+    @Override
+    public void process(ServiceNode serviceNode, List<AnnotationAttachmentNode> annotations) {
+        List<BLangFunction> resources = (List<BLangFunction>) serviceNode.getResources();
+        resources.forEach(res -> validate(serviceNode.getName().getValue(), res, this.diagnosticLog));
+//        if (resourceCount != 4) {
+//            String msg = String
+//                    .format("Service needs to have all 4 resources[(%s or %s), %s, %s, %s].", RESOURCE_ON_ACCEPT,
+//                            RESOURCE_ON_CONNECT, RESOURCE_ON_READ_READY, RESOURCE_ON_CLOSE, RESOURCE_ON_ERROR);
+//            diagnosticLog.logDiagnostic(ERROR, serviceNode.getPosition(), msg);
+//        }
```

Comment:
Why have we commented out these lines?

Reply:
There is fix pending from the core team.